add option to show pagination on top of search results

orthagh opened the pull request:
Requester after an early internal review of 11.0 UI.


![image](https://github.com/glpi-project/glpi/assets/418844/1b2dc15c-d9b1-46ee-b109-f5b1bf2951ca)

AdrienClairembault commented:
Maybe the items should be centered ? 

![image](https://github.com/glpi-project/glpi/assets/42734840/6e4aae26-c80f-48c2-ada9-066b2e238bfe)

This make the header feel "crowded" but since it is not the default option I am OK with it.

cconard96 commented:
This conflicts with the new status message area for mail collectors and automatic actions.

![Selection_279](https://github.com/glpi-project/glpi/assets/17678637/56464e6b-f79c-4be7-af0b-9cdd055aa72e)

orthagh commented:
> This conflicts with the new status message area for mail collectors and automatic actions.

I switched the position of the two blocks, it should be better (or at least, not so bad). As we have only receivers with this kind of display, I guess it's ok. Alerts in the header are not ideal, but for the moment any alternative was worse (double scrollbars for example).

> Maybe the items should be centered ?

Done

> This make the header feel "crowded"

I tried with the pager between the global header and the columns header, it's harder to achieve with 3 stickies and it makes the space reserved for headers a bit too much.
I take any alternative suggestions.

cconard96 commented:
> I take any alternative suggestions.
Maybe the status alerts shouldn't be shown in the search pages at all. We have a message area on the central homepage, even if it is currently used for post-install/post-update messages, they could fit in there. Although, that could be an issue for people who hide the central messages to ignore the optional migration warnings.

The only other suggestion I have is a lot more work and a new feature where there is a sort of notification area/inbox along the top of the page next to the user dropdown where these kinds of alerts can be shown.

AdrienClairembault commented:
Spacing feels a bit weird:

![image](https://github.com/glpi-project/glpi/assets/42734840/5a702f8f-dfe2-448a-8109-34659cf21480)

Shouldn't (1) and (2) be the same size ?

AdrienClairembault commented:
And the number of rows can't be changed anymore:

![row_number](https://github.com/glpi-project/glpi/assets/42734840/b1b1e7c9-7918-4934-b1da-7a026585e7a6)

It works fine without the option, definitely related to these changes.

orthagh commented:
> Spacing feels a bit weird:

Fixed

> And the number of rows can't be changed anymore

fixed

Current iteration: 
![image](https://github.com/glpi-project/glpi/assets/418844/7aa00991-c808-43b5-866a-0763c935a6f9)

cconard96 commented on templates/components/pager.html.twig:
```diff
@@ -72,20 +74,26 @@
-    <span class="search-limit d-block d-md-none">
+    {% if not short_display %}
+        <span class="search-limit d-none d-md-block">
+            {{ __('%s rows / page')|format(limitdropdown)|raw }}
```
```suggestion
            {{ __('%s rows / page')|e|format(limitdropdown)|raw }}
```
I think if you format a localization string and then output as raw, you need to escape the text just before formatting in the rare case that a translation can have HTML characters.

AdrienClairembault reviewed (changes requested):
I am not sure if it is related to your changes but this button has a very thin box shadow that should be removed: 
![image](https://github.com/glpi-project/glpi/assets/42734840/f6b7fb6e-038d-4127-b635-7d4522f72ec1).

The pagination actions are not aligned correctly:
![image](https://github.com/glpi-project/glpi/assets/42734840/0e145248-7675-4537-b63d-e9462901e65d)


You need to add "display: flex" to each `<a>` tags:
![image](https://github.com/glpi-project/glpi/assets/42734840/a3d096e7-7754-4963-aa89-0f82cd4d4c40)

orthagh commented:
Done

cedric-anne commented on install/mysql/glpi-empty.sql:
```diff
@@ -7825,6 +7825,7 @@ CREATE TABLE `glpi_users` (
   `password_history` longtext,
   `is_notif_enable_default` tinyint DEFAULT NULL,
   `show_search_form` tinyint DEFAULT 0,
+  `search_pagination_on_top` tinyint DEFAULT 0,
```
Default value for users should be `null`. Indeed, this `null` value will permit to fallback to the value configured globally in the `glpi_configs` table.

```suggestion
  `search_pagination_on_top` tinyint DEFAULT NULL,
```

By the way, it looks like the `show_search_form` default value should be fixed.

trasher commented on install/migrations/update_10.0.x_to_11.0.0/user.php:
```diff
@@ -46,7 +46,9 @@
 
 // search ux improvements (#15861)
 $migration->addField('glpi_users', 'show_search_form', "tinyint DEFAULT 0");
+$migration->addField('glpi_users', 'search_pagination_on_top', "tinyint DEFAULT 0");
```
```suggestion
$migration->addField('glpi_users', 'search_pagination_on_top', "tinyint DEFAULT NULL");
```

cedric-anne commented on install/migrations/update_10.0.x_to_11.0.0/user.php:
```diff
@@ -46,7 +46,9 @@
 
 // search ux improvements (#15861)
 $migration->addField('glpi_users', 'show_search_form', "tinyint DEFAULT 0");
```
```suggestion
$migration->addField('glpi_users', 'show_search_form', "tinyint DEFAULT NULL");
```

cedric-anne commented on install/mysql/glpi-empty.sql:
```diff
@@ -7825,6 +7825,7 @@ CREATE TABLE `glpi_users` (
   `password_history` longtext,
   `is_notif_enable_default` tinyint DEFAULT NULL,
   `show_search_form` tinyint DEFAULT 0,
```
```suggestion
  `show_search_form` tinyint DEFAULT NULL,
```
